extensions/elasticsearch/elasticsearch-5/src/test/java/com/hazelcast/jet/elastic/impl/ElasticSourcePTest.java

```diff
@@ -0,0 +1,276 @@
+        // rather than in setUp()
+        processor = new ElasticSourceP<>(configuration, shards);
+
+        return TestSupport.verifyProcessor(() -> processor)
+                .disableSnapshots();
+    }
+
+    @Test
+    public void when_runProcessor_then_executeSearchRequestWithScroll() throws Exception {
+        when(response.getHits()).thenReturn(new SearchHits(new SearchHit[]{}, 0, Float.NaN));
+
+        TestSupport support = runProcessor();
+
+        support.expectOutput(emptyList());
+
+        ArgumentCaptor<SearchRequest> captor = forClass(SearchRequest.class);
+        verify(spyClient).search(captor.capture());
+
+        SearchRequest request = captor.getValue();
+        assertThat(request.scroll().keepAlive().getStringRep()).isEqualTo(KEEP_ALIVE);
+    }
+
+    @Test
+    public void given_singleHit_when_runProcessor_then_produceSingleHit() throws Exception {
+        SearchHit hit = new SearchHit(0, "id-0", new Text("ignored"), emptyMap());
+        hit.sourceRef(new BytesArray(HIT_SOURCE));
+        when(response.getHits()).thenReturn(new SearchHits(new SearchHit[]{hit}, 1, Float.NaN));
+
+        SearchResponse response2 = mock(SearchResponse.class);
+        when(response2.getHits()).thenReturn(new SearchHits(new SearchHit[]{}, 1, Float.NaN));
+        doReturn(response2).when(spyClient).searchScroll(any());
+
+        TestSupport testSupport = runProcessor();
+
+        testSupport.expectOutput(newArrayList(HIT_SOURCE));
+    }
+
+    @Test
+    public void givenMultipleResults_when_runProcessor_then_useScrollIdInFollowupScrollRequest() throws Exception {
+        SearchHit hit = new SearchHit(0, "id-0", new Text("ignored"), emptyMap());
+        hit.sourceRef(new BytesArray(HIT_SOURCE));
+        when(response.getHits()).thenReturn(new SearchHits(new SearchHit[]{hit}, 3, Float.NaN));
+
+        SearchResponse response2 = mock(SearchResponse.class);
+        SearchHit hit2 = new SearchHit(1, "id-1", new Text("ignored"), emptyMap());
+        hit2.sourceRef(new BytesArray(HIT_SOURCE2));
+        when(response2.getHits()).thenReturn(new SearchHits(new SearchHit[]{hit2}, 3, Float.NaN));
+
+        SearchResponse response3 = mock(SearchResponse.class);
+        when(response3.getHits()).thenReturn(new SearchHits(new SearchHit[]{}, 3, Float.NaN));
+        doReturn(response2, response3).when(spyClient).searchScroll(any()); // client is a spy, need to to doReturn()..
+
+        TestSupport testSupport = runProcessor();
+
+        testSupport.expectOutput(newArrayList(HIT_SOURCE, HIT_SOURCE2));
+
+        ArgumentCaptor<SearchScrollRequest> captor = forClass(SearchScrollRequest.class);
+
+        verify(spyClient, times(2)).searchScroll(captor.capture());
+        SearchScrollRequest request = captor.getValue();
+        assertThat(request.scrollId()).isEqualTo(SCROLL_ID);
+        assertThat(request.scroll().keepAlive().getStringRep()).isEqualTo(KEEP_ALIVE);
+    }
+
+/*
+    @Test
```

Comment:
why is this commented out ?

Reply:
Because the API is too different for this test to work in the way it was designed and I considered writing new test would not provide much benefit for v5. 

Deleted commented code.